src/cli/java/org/commcare/util/screen/EntityListSubscreen.java

```diff
@@ -45,24 +49,64 @@ public class EntityListSubscreen extends Subscreen<EntityScreen> {
     private List<Entity<TreeReference>> entities;
 
     public EntityListSubscreen(Detail shortDetail, Vector<TreeReference> references, EvaluationContext context,
-            boolean handleCaseIndex) throws CommCareSessionException {
+            boolean handleCaseIndex, EntityScreenContext entityScreenContext) throws CommCareSessionException {
         mHeader = createHeader(shortDetail, context);
         this.shortDetail = shortDetail;
         this.rootContext = context;
         this.entitiesRefs = new TreeReference[references.size()];
         this.handleCaseIndex = handleCaseIndex;
+        this.entityScreenContext = entityScreenContext;
         references.copyInto(entitiesRefs);
         actions = shortDetail.getCustomActions(context);
-        initEntities(context, shortDetail);
+        initEntities(context, shortDetail, entityScreenContext);
     }
 
-    private void initEntities(EvaluationContext context, Detail shortDetail) {
+    private void initEntities(EvaluationContext context, Detail shortDetail,
+            EntityScreenContext entityScreenContext) {
         NodeEntityFactory nodeEntityFactory = new NodeEntityFactory(shortDetail, context);
         entities = new ArrayList<>();
         for (TreeReference reference : entitiesRefs) {
             entities.add(nodeEntityFactory.getEntity(reference));
         }
         nodeEntityFactory.prepareEntities(entities);
+        filterEntities(entityScreenContext);
+        sortEntities(entityScreenContext);
+    }
+
+    private void filterEntities(EntityScreenContext entityScreenContext) {
+        String searchText = entityScreenContext.getSearchText();
+        boolean isFuzzySearchEnabled = entityScreenContext.isFuzzySearch();
+        if (searchText != null && !"".equals(searchText)) {
+            NodeEntityFactory nodeEntityFactory = new NodeEntityFactory(shortDetail, rootContext);
+            EntityStringFilterer filterer = new EntityStringFilterer(searchText.split(" "),
+                    nodeEntityFactory, entities, isFuzzySearchEnabled);
```

Comment:
this class doesn't appear to actually use the `nodeEntityFactory` for anything except checking `nodeFactory.isEntitySetReady()` which could be moved out. Also, since this is a new instance of `NodeEntityFactory` I think this will raise a runtime exception because `prepareEntities` hasn't been called (but that's moot if we remove the need for it in the first place).